src/deadline/unreal_logger/logger.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import logging
+
+
+from deadline.unreal_logger.handlers import UnrealLogHandler
+
+
+def get_logger() -> logging.Logger:
+    """
+    Returns an instance of logging.Handler.
+    Attach handler :class:`deadline.unreal_logger.handlers.UnrealLogHandler` if unreal module is
+    available
+    """
+
+    unreal_logger = logging.getLogger("unreal_logger")
+    print(id(unreal_logger))
+    unreal_logger.setLevel(logging.DEBUG)
+
+    try:
+        import unreal  # noqa: F401
+    except ModuleNotFoundError:
+        unreal = None
+
+    # can be called outside UE so need to check before adding UE specific handler
+    if unreal is not None:
+        unreal_log_handler = UnrealLogHandler(unreal)
+        unreal_log_handler.setLevel(logging.DEBUG)
+
+        # If we found existed UnrealLogHandler, skip adding duplicate
+        existed_unreal_log_handler = next(
```

Comment:
Much cleaner now!  Only thing I'm wondering is whether this could all just basically be an "initialized" bool we set after we've done it once, or is there something more complex we're handling here?

Reply:
you mean using global `initialized` bool in try except block and check for it instead of `unreal is not None`?

Reply:
Theoretically, the module unreal is either available or not (run in session or outside of it). Therefore, we can move the import to the top of the file and set the variable. But I still think the check for the presence of the handler should be left

Smth like this:
```
try:
    import unreal  # noqa: F401
    unreal_initialized = True
except ModuleNotFoundError:
    unreal = None
    unreal_initialized = False


def get_logger() -> logging.Logger:
    unreal_logger = logging.getLogger("unreal_logger")
    unreal_logger.setLevel(logging.DEBUG)
    
    # can be called outside UE so need to check before adding UE specific handler
    if unreal_initialized:
        ...
```

Reply:
Well once we've called get_logger once and either added the handler or not added the handler we should be able to use the bool to know if we need to potentially add it again I think?

Reply:
Good suggestion, but we still need to know the first time if there is a suitable unreal handler. So iteration with next is unavoidable, I think

Reply:
So lets do this way for example
```
try:
    import unreal  # noqa: F401

    UNREAL_INITIALIZED = True
except ModuleNotFoundError:
    unreal = None
    UNREAL_INITIALIZED = False


UNREAL_HANDLER_ADDED = False

def get_logger() -> logging.Logger:
    unreal_logger = logging.getLogger("unreal_logger")
    unreal_logger.setLevel(logging.DEBUG)

    global UNREAL_HANDLER_ADDED

    # can be called outside UE so need to check before adding UE specific handler
    if not UNREAL_HANDLER_ADDED and UNREAL_INITIALIZED:
       # just add handler
       ...
```